File: Module12/Shivangi_resource/Module12_RESTfulAPI_Projects/Module12_RESTfulAPI_Projects/student-api-express/routes/students.js
const express = require("express");
const router = express.Router();

// In-memory data (same as Python version)
let students = [
  { id: 1, name: "Anita", course: "Python" },
  { id: 2, name: "Rahul", course: "Data Science" }
];

// GET all students
router.get("/", (req, res) => {
  res.json(students);
});

// POST add student
router.post("/", (req, res) => {
  const newStudent = {
    id: students.length ? students[students.length - 1].id + 1 : 1,
    name: req.body.name,
    course: req.body.course
  };

  students.push(newStudent);
  res.status(201).json(newStudent);
});

// PUT update student
router.put("/:id", (req, res) => {
  const id = parseInt(req.params.id);

  const student = students.find(s => s.id === id);

  if (!student) {
    return res.status(404).json({ message: "Student not found" });
  }

  student.name = req.body.name ?? student.name;
  student.course = req.body.course ?? student.course;

  res.json(student);
});

// DELETE student
router.delete("/:id", (req, res) => {
  const id = parseInt(req.params.id);

  const index = students.findIndex(s => s.id === id);

  if (index === -1) {
    return res.status(404).json({ message: "Student not found" });
  }

  students.splice(index, 1);
  res.json({ message: "Student deleted" });
});

module.exports = router;

File: Module12/Shivangi_resource/Module12_RESTfulAPI_Projects/Module12_RESTfulAPI_Projects/studentsApiFlask/students.py
students = [
    {"id":1,"name":"John","course":"Python"},
    {"id":2,"name":"Harry","course":"Java"}
]

def get_all_students():
    return students

def add_student(body):
    body["id"] = students[-1]["id"]+1
    students.append(body)
    return body,201

def update_student(id,body):
    for student in students:
        if student["id"] == id:
            student.update(body)
            return student
    return {"message":"Student not found"},404

def delete_student(id):
    for student in students:
        if student["id"] == id:
            students.remove(student)
            return {"message":"Student deleted"}
    return {"message":"Student not found"},404
        
 
